src/armnn.js

```diff
@@ -0,0 +1,636 @@
+/* jshint esversion: 6 */
+/* eslint "indent": [ "error", 4, { "SwitchCase": 1 } ] */
+
+var armnn = armnn || {};
+var base = base || require('./base');
+var flatbuffers = flatbuffers || require('flatbuffers').flatbuffers;
+var long = long || { Long: require('long') };
+
+armnn.ModelFactory = class {
+
+    match(context) {
+        const extension = context.identifier.split('.').pop().toLowerCase();
+        if (extension == 'armnn') {
+            return true;
+        }
+        return false;
+    }
+
+    open(context, host) {
+        return host.require('./armnn-schema').then((armnn_schema) => {
+            const identifier = context.identifier;
+            let model = null;
+            try {
+                const buffer = context.buffer;
+                const byteBuffer = new flatbuffers.ByteBuffer(buffer);
+                armnn.schema = armnn_schema.armnnSerializer;
+                model = armnn.schema.SerializedGraph.getRootAsSerializedGraph(byteBuffer);
+            }
+            catch (error) {
+                host.exception(error, false);
+                let message = error && error.message ? error.message : error.toString();
+                message = message.endsWith('.') ? message.substring(0, message.length - 1) : message;
+                throw new armnn.Error(message + " in '" + identifier + "'.");
+            }
+
+            return armnn.Metadata.open(host).then((metadata) => {
+                try {
+                    return new armnn.Model(model, metadata);
+                }
+                catch (error) {
+                    let message = error && error.message ? error.message : error.toString();
+                    message = message.endsWith('.') ? message.substring(0, message.length - 1) : message;
+                    throw new new armnn.Error(message + " in '" + identifier + "'.");
+                }
+            });
+        });
+    }
+};
+
+armnn.Model = class {
+
+    constructor(model, metadata) {
+        this._format = 'ArmNN';
```

Comment:
Says "Arm NN" on the website?

Reply:
I couldn't find format identifier in website (and also in code)...

Anyway, not format but file identifier is defined as 'ARMN' in flatbuffers scheme.

https://github.com/ARM-software/armnn/blob/branches/armnn_19_08/src/armnnSerializer/ArmnnSchema.fbs

Reply:
This is not used as an identifier but shown to the user in the Properties dialog. Should use the same style as the `README.md` on https://github.com/ARM-software/armnn.